index.js

```diff
@@ -79,12 +80,15 @@ async function handleRedemption ($redemptionContainer) {
     const redemptionData = await extractAllData($redemptionContainer)
     log("DEBUG redemptionData", redemptionData)
     try {
```

Comment:
I propose this
```
    try {
        if (!rewards[redemptionData.rewardName]) {
            return
        }
        const result = await rewards[redemptionData.rewardName](redemptionData)
        if (result) {
            redemptionData.actions.resolve.click()
        } else {
            redemptionData.actions.reject.click()
        }
    } catch (e) {
        // this means something really went wrong, maybe we should let the streamer decide
        console.error(e.message)
    }
```

Reply:
Done! We now have something similar, based on our idea of having a success boolean.
```javascript
const rewardFunction = rewards[redemptionData.rewardName]
if (rewardFunction) {
    try {
        const result = await rewards[redemptionData.rewardName](redemptionData)
        if (result.success) {
            log(result.message)
            redemptionData.actions.resolve.click()
        } else {
            log(result.message)
            redemptionData.actions.reject.click()
        }
    } catch (e) {
        // unexpected reward failure!
        console.error(e.message)
        redemptionData.actions.reject.click()
    }
} else {
    // don't do anything with unhandled redemptions
    log("Received unhandled reward:", redemptionData.rewardName + ", ignoring")
}
```